tests/test_cuda_targeted_insertion_mover.py

```diff
@@ -25,13 +26,23 @@
 from timemachine.potentials import HarmonicBond, Nonbonded, SummedPotential
 
 
-def compute_ref_raw_log_prob(ref_exchange, water_idx, vi_mols, vj_mols, vol_i, vol_j, coords, box, new_coords):
+def compute_ref_raw_log_prob(
+    ref_exchange, water_idx, vi_mols, vj_mols, vol_i, vol_j, coords, box, sampled_mol_idx, new_coords
+):
     """Modified from timemachine.md.exchange.exchange_mover.TIBDExchangeMove.swap_vi_into_vj
     to support not resampling"""
     coords = coords.copy()
     log_weights_before_full = ref_exchange.batch_log_weights(coords, box)
     log_weights_before = log_weights_before_full[vi_mols]
 
+    # Verify that the probability of moving the specific water is not highly unlikely
+    src_idx = np.argwhere(vi_mols == sampled_mol_idx).reshape(-1)
+    log_probs_before = log_weights_before - logsumexp(log_weights_before)
+    probs_before = np.exp(log_probs_before)
+    assert (
```

Comment:
Need to refine this some more

Reply:
Does it make sense to test in log_weights?

Reply:
Not sure what you mean. This takes the 'before' log weights (or the source region weights) computes the normalized probability and then evaluates that rather than the log weights.

Do you mean that it is testing within this function? If that is what you mean that would be doable, though would require duplicating the logic of constructing the before log weights to do so which is why I put it here.

Reply:
(discussed offline, point of confusion from me)